src/common/io.py

```diff
@@ -1477,6 +1475,123 @@ def get_variable_data(self, name):
         else:
             return Trajectory(self._get_diagnostics_trajectory(0),factor*self._get_interpolated_trajectory(dataInd))
 
+    def get_variables_data(self,
+                           names: list[str],
+                           start_index: int = 0,
+                           stop_index: Union[int, None] = None
+    ) -> tuple[list[Trajectory], Union[int, None]]:
+        """"
+            Returns multiple trajectories, sliced to index range.
+            Note that start_index and stop_index behaves as indices for slicing, i.e. array[start_index:stop_index].
+            This also implies that stop_index = None or stop_index larger than the number of available data points
+            results in retrieving all the available data points from start_index, i.e. as the slice [start_index:].
+
+            Parameters::
+
+                names --
+                    List of variables names for which to fetch trajectories.
+
+                start_index --
+                    Starting index for trajectory slicing.
+
+                stop_index --
+                    Stopping index for trajectory slicing.
+
+            Returns::
+                Tuple: (List of trajectories, next start index (non-negative))
+        """
+
+        """
+            TODO:
+            For [start_index, stop_index] we can do performance improvements
+            since methods such as _get_trajectory, get_diagnostics_trajectory
+            and _get_interpolated_trajectory always retrieve the full
+            trajectory which is then sliced, here. Instead we can account for
+            start_index and stop_index and reduce the time spent reading unused data points.
+        """
+        trajectories = []
+
```

Comment:
Check that stop_index is > than start_index?

Reply:
I did have that initially but I removed that check with the following reason: If we want `start_index` and `stop_index` to behave like Python-slices i.e. `[start_index:stop_index]`, then we should not check that, since `array[5:0]` does not raise an error in Python for arbitrary arrays/lists.

Reply:
You are right that it this follows the Python "standard" for slicing. However, I do not think it makes sense to make a call with stop_index < start_index and thus should result in error

Reply:
Fixed